### ISSUE TITLE
dpDayPicker on FormControl (reactive form) initially set as dirty or (pristine = false)

### ISSUE THREAD

**Issue by whoang11** [vlio20/angular-datepicker#112]
The dpDayPicker on FormControl was initially set to dirty or (pristine = false). As a workaround,  I had to set this control's pristine to true after creating the form group.  Attached is the picture file that has the view and code for this issue:
![dpdaypickerissue](https://user-images.githubusercontent.com/7757548/27597042-da8df51e-5b26-11e7-8604-b848226b88d3.PNG)

As  seen on the picture, I also used other controls. However, these controls' pristine  were initially set to true

Thank you sharing your dpDayPicker.

Wen H.

**Comment by vlio20** [comment]
@whoang11, thanks. will check it out.

**Comment by vlio20** [comment]
@whoang11, can post here your code - not as a screenshot?

**Comment by whoang11** [comment]
Here are my view and code:

 <div class="form-group" [ngClass]="{'has-error': displayMessage.appointmentDate }">
      <label>Appointment Date*</label>
      <i tooltip="{{maxDate}}" placement="right" class="blueColorInfo fa fa-info-circle"></i>
      <em class="validationMs" *ngIf="displayMessage.appointmentDate"> {{ displayMessage.appointmentDate }} </em>
      <em class="validationMs" *ngIf="mouseoverSubmit && !displayMessage.appointmentDate &&
                appointmentForm.controls.appointmentDate.errors?.required">{{validationMessages.appointmentDate.required }}</em>
      <input class="form-control" id="appointmentDate"  formControlName="appointmentDate" [dpDayPicker]="datePickerConfig" theme="dp-material"/>
    </div>

 ngOnInit() {

    // const today = new Date();
    this.appointmentForm = this.fb.group({
      customerName: ['', [Validators.required, Validators.maxLength(50)]],
      phone: ['', [Validators.required, Validators.pattern('^[0-9]{3}-[0-9]{3}-[0-9]{4}?$')]],
      streetAddress: ['', Validators.required],
      city: ['', Validators.required],
      stateId: ['', Validators.required],
      zipCode: ['', [Validators.required, Validators.pattern('^[0-9]{5}(?:-[0-9]{4})?$')]],
      appointmentDate: ['',  Validators.required],
      appointmentTime: ['', Validators.required],

    });

    setTimeout(() =>  {
         this.appointmentForm.get('appointmentDate').markAsPristine(true);
    }, 10);

  }

**Comment by whoang11** [comment]
Here is the view that was not displayed correctly above:

 <div class="form-group" [ngClass]="{'has-error': displayMessage.appointmentDate }">
      <label>Appointment Date*</label>
      <i tooltip="{{maxDate}}" placement="right" class="blueColorInfo fa fa-info-circle"></i>
      <em class="validationMs" *ngIf="displayMessage.appointmentDate"> {{ displayMessage.appointmentDate }} </em>
      <em class="validationMs" *ngIf="mouseoverSubmit && !displayMessage.appointmentDate &&
                appointmentForm.controls.appointmentDate.errors?.required">{{validationMessages.appointmentDate.required }}</em>
      <input class="form-control" id="appointmentDate"  formControlName="appointmentDate" [dpDayPicker]="datePickerConfig" theme="dp-material"/>
    </div>

**Comment by vlio20** [comment]
Found the issue, will soon push a fix. 
@whoang11 thanks for pointing out!

**Comment by whoang11** [comment]
I upgraded to the version 2.1.1. This version has  fixed the issue of  setting the dpDayPicker to dirty initially. However, this version has a new problem of not setting the dpDayPicker to dirty when I used the mouse to select a different date. 

If I typed in the date, the dpDayPicker was correctly  set to dirty.

For this new problem, I went back to version 1.4.3 and continued using this work around solution:
setTimeout(() =>  {
     this.appointmentForm.get('appointmentDate').markAsPristine(true);
}, 10);

Thanks,
